### DISCUSSION TITLE
Sunsetting Projects (classic) breaks existing functionality

### DISCUSSION THREAD

**Discussion by slinna** [community/community#127081] [Projects and Issues]
### Select Topic Area

Bug

### Feature Area

Projects

### Body

Now that GitHub is sunsetting the legacy Projects (https://github.blog/changelog/2024-05-23-sunset-notice-projects-classic/), I would like to point out that this move will break existing functionality that we have implemented for several of our customers. The problem is that there is no API-level feature parity, as the ProjectsV2 lacks the issue history as pointed out in these forum discussions: 

https://github.com/orgs/community/discussions/39668
https://github.com/orgs/community/discussions/5859

We are using GraphQL as per your recommendation at the time we started working with your API. But it's still not providing the essential functionality for ProjectsV2 that our customers have been using for years with the Projects (classic). 

Is it too much to ask you to implement the issue history for the new Projects before the forced migration on August 23?  

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by slinna** [comment]
@evi-liu, that's great news, I'm looking forward to hearing more updates!

**Comment by a6uxa4** [comment]


**Reply by su1i4** [reply]
helo0 a6uxa4

**Reply by a6uxa4** [reply]
unmark as answer dep chykan jokpu

**Comment by slinna** [comment]
@evi-liu, is there any ETA for when the GraphQL API support for the issue history will be available? The forced projects v2 migration is now one month away, and we'd need some time to implement the changes before our customers' existing functionality breaks.

**Reply by evi-liu** [reply]
Hi @slinna, apologies that my previous comment was accidentally deleted. 

After looking into whether we could support parity for `MovedColumnsInProjectEvent`, we realized it would be a non-trivial effort. However, there is potential that we could support something like `MovedStatusInProjectEvent`, where you could see if a status was changed - would that be sufficient for your use case?

**Reply by slinna** [reply]
@evi-liu Yes, if we can get the history of the status changes of an issue or a pull request, that would allow us to implement the functionality we currently have for classic projects.

**Reply by evi-liu** [reply]
@slinna Thank you, that's very helpful to know. I'll see what we can do to get this available publicly in the near future.

**Reply by slinna** [reply]
@evi-liu is there any ETA for when the GraphQL API support for the status changes history will be available?

**Reply by evi-liu** [reply]
Hi @slinna! Some good news - we should be able to make this GraphQL event publicly available in late September/early October!

**Reply by slinna** [reply]
@evi-liu that's great news, looking forward to it!

**Reply by slinna** [reply]
@evi-liu, trying to do some planning here. Is your previous ETA for late September/early October still valid?

**Reply by evi-liu** [reply]
Hi @slinna, I just checked, and while we were very close to releasing this, we unfortunately encountered a couple of blockers at the end. I’ll work to get it prioritized as soon as possible, though it’s unlikely to be ready this month. Apologies for the missed ETA.

**Reply by slinna** [reply]
@evi-liu do you have a new ETA to share?

**Reply by jrousseau-gie-iris** [reply]
Do you have any news regarding the fix date?

**Reply by rileybroughten** [reply]
You'll now see additional events in GraphQL including `Status` changes on an issue. Check out the [changelog](https://github.blog/changelog/2025-11-06-improved-onboarding-flow-for-github-projects/#%f0%9f%a4%96-api-improvements) for more details!

**Reply by slinna** [reply]
@rileybroughten that's great news, we'll give it a try!

**Comment by Xelliese** [comment]
If immediate resolution isn’t feasible, consider creating a custom solution that uses GitHub’s API to replicate the issue history functionality. For example, you can use the GitHub GraphQL API to query and aggregate issue data from the classic projects into a separate database or system.
